File: pr1-java/client/src/main/java/pr1Java/client/javafx/reflection/clients/Window.java
package pr1Java.client.javafx.reflection.clients;

import pr1Java.client.Configuration;
import pr1Java.model.User;
import pr1Java.services.reflection.ReflectionServices;

public abstract class Window {
    protected ReflectionServices services = null;
    protected User signedInUser = null;

    public void init(ReflectionServices services, User signedInUser) {
        Configuration.logger.traceEntry("entering init with {} and {}", services, signedInUser);

        this.services = services;
        this.signedInUser = signedInUser;

        Configuration.logger.traceExit();
    }
}

File: pr1-cs/Client/Gtk/Reflection/Clients/Window.cs
using Gtk;
using Model;
using Services;
using Services.Reflection;

namespace Client.Gtk.Reflection.Clients
{
    public abstract class Window
    {
        protected readonly Builder GuiElements = new Builder();
        protected IServices Services;
        protected User SignedInUser;
        protected global::Gtk.Window OwnedWindow;

        public virtual Window Init(IServices services, User signedInUser)
        {
            ClientStarter.OpenWindows++;
            Services = services;
            SignedInUser = signedInUser;

            return this;
        }

        public void Open()
        {
            OwnedWindow.Show();
        }

        protected void Close()
        {
            ClientStarter.OpenWindows--;
            if (ClientStarter.OpenWindows <= 0)
                Application.Quit();
            OwnedWindow.Destroy();
        }
    }
}